bus-mapping/src/circuit_input_builder.rs

```diff
@@ -1325,16 +1632,17 @@ impl<P: JsonRpcClient> BuilderClient<P> {
         sdb: StateDB,
         code_db: CodeDB,
         eth_block: &EthBlock,
+        history_hashes: Vec<Word>,
```

Comment:
This will always be of length = 256 right?

Reply:
It could be less when current block number <= 256.

Reply:
I see.  In that case, I believe it's preferrable (for consistency) that geth-util tracer should behave the same way with the `HistoryHashes` it takes as input., and currently it does this:
```go
		GetHash: func(n uint64) common.Hash {
			number := block.Number.ToInt().Uint64()
			if number > n && number-n <= 256 {
				return common.BigToHash(toBigInt(block.HistoryHashes[256-number+n]))
			}
			return common.Hash{}
		}
```
If number = 10, then len(HistoryHashes) = 9.  Say that we query n = 9.  This is `256 - 10 + 9 = 255` which is out of bounds for a slice of length 9.

I'd suggest changing it to `block.HistoryHashes[len(block.HistoryHashes)-number+n]`.
Or redefining HistoryHashes to always be length 256, and for "negative numbers" set the zero hash.

Reply:
You are right! Let me fix it.